sigopt/xgboost/experiment.py

```diff
@@ -3,30 +3,41 @@
 import copy
 
 from .. import create_experiment
+from .defaults import (
+  SEARCH_BOUNDS,
+  SEARCH_PARAMS,
+  DEFAULT_SEARCH_PARAMS,
+  DEFAULT_BO_ITERATIONS,
+  DEFAULT_NUM_BOOST_ROUND,
+  DEFAULT_REGRESSION_METRICS,
+  DEFAULT_CLASSIFICATION_METRICS,
+  SUPPORTED_METRICS_TO_OPTIMIZE
+)
 
-DEFAULT_CLASSIFICATION_METRICS = ['accuracy', 'precision', 'recall', 'F1']
-DEFAULT_REGRESSION_METRICS = ['mean absolute error', 'mean squared error']
-DEFAULT_NUM_BOOST_ROUND = 10
 
 class XGBExperiment:
   def __init__(self, experiment_config, dtrain, evals, params, num_boost_round, run_options):
     self.experiment_config = experiment_config
+    self.experiment_config_parsed = None
     self.dtrain = dtrain
     self.evals = evals
     self.params = params
     self.num_boost_round = num_boost_round
     self.run_options = run_options
     self.sigopt_experiment = None
 
-  def parse_and_create_parameter_space(self):
-    # TODO: deal with later
-    pass
-
-  def parse_and_create_experiment(self):
-    experiment_config_parsed = copy.deepcopy(self.experiment_config)
+  def parse_and_create_metrics(self):
```

Comment:
Do we want to give user the option of not supplying the metrics?

Reply:
Okay I will figure out from the objective (accuracy / RMSE as defaults), and it's another task I'll throw an error